src/main.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashMap;
+use std::env;
+use std::num::ParseIntError;
+use std::sync::Mutex;
+
+use actix_web::http;
+use actix_web::middleware::cors::Cors;
+use actix_web::{server, App, HttpRequest, HttpResponse, Json};
+use sentry::integrations::failure::capture_error as capture_failure_error;
+use sentry::protocol::value::to_value;
+use sentry::User;
+use sentry_actix::SentryMiddleware;
+use serde::{Deserialize, Serialize};
+
+lazy_static::lazy_static! {
+    static ref HASHMAP: Mutex<HashMap<&'static str, u32>> = {
+        let mut inventory = HashMap::new();
+        inventory.insert("wrench", 1);
+        inventory.insert("nails", 1);
+        inventory.insert("hammer", 1);
+        Mutex::new(inventory)
+    };
+}
+
+fn multiply_new(first_number_str: &str, second_number_str: &str) -> Result<i32, ParseIntError> {
+    let first_number: i32 = first_number_str.parse()?;
+    let second_number: i32 = second_number_str.parse()?;
+    Ok(first_number * second_number)
+}
+
+fn handled_new(_req: &HttpRequest) -> String {
+    let first = "t";
+    let second = "2";
+    match multiply_new(first, second) {
+        Ok(result) => format!("{} * {} => {}", first, second, result),
+        Err(err) => {
+            sentry::capture_error(&err);
+            "try again".to_string()
+        }
+    }
+}
+
+fn fakedatabseapp(_req: &HttpRequest) -> HttpResponse {
+    panic!("Unhandled request!");
+}
+
+#[derive(Deserialize, Clone, Debug)]
+struct CardSubmittedPayload {
+    card_id: i64,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct Item {
+    id: String,
+    name: String,
+    price: f64,
+    img: String,
+}
+
+#[derive(Serialize, Clone, Debug, Deserialize)]
+struct CheckoutPayload {
+    email: String,
+    cart: Vec<Item>,
+}
+
+fn process_order(cart: &[Item]) -> String {
+    let mut map = HASHMAP.lock().unwrap();
+    println!("The entry for `0` is \"{:?}\".", map.get("foo"));
+
+    for cartitem in cart.iter() {
+        match map.get_mut(cartitem.id.as_str()) {
+            Some(stock) if *stock > 0 => {
+                *stock -= 1;
+                println!(
+                    "Success: {} was purchased, remaining stock is {}",
+                    cartitem.id, stock
+                );
+            }
+            _ => {
+                sentry::configure_scope(|scope| {
+                    scope.set_extra("inventory", to_value(&*map).unwrap());
+                });
+
+                let message = format!("Not enough inventory for {}", cartitem.id);
+                capture_failure_error(&failure::format_err!("Error: {}", message));
```

Comment:
if there's not enough inventory, we need to return a 500 response. I believe that's the only remaining ToDo on this PR